commands/proposal_submission.go

```diff
@@ -375,35 +379,60 @@ func checkOracleSpec(spec *oraclespb.OracleSpecConfiguration, name string) Error
 				errs.AddForProperty(fmt.Sprintf("proposal_submission.terms.change.new_market.changes.instrument.product.future."+name+".pub_keys.%d", i), ErrIsNotValid)
 			}
 		}
-		if len(spec.Filters) == 0 {
-			errs.AddForProperty("proposal_submission.terms.change.new_market.changes.instrument.product.future."+name+".filters", ErrIsRequired)
+
+		return checkOracleSpecFilters(spec, name, errs)
+	} else {
+		errs.AddForProperty("proposal_submission.terms.change.new_market.changes.instrument.product.future."+name, ErrIsRequired)
+	}
+
+	return errs
+}
+
+func isBuiltInSpec(filters []*oraclespb.Filter) bool {
+	if len(filters) != 1 {
+		return false
+	}
+
+	if filters[0].Key == nil || filters[0].Conditions == nil {
+		return false
+	}
+
+	if strings.HasPrefix(filters[0].Key.Name, "vegaprotocol.builtin") && filters[0].Key.Type == oraclespb.PropertyKey_TYPE_TIMESTAMP {
+		return true
+	}
+
+	return false
+}
+
+func checkOracleSpecFilters(spec *oraclespb.OracleSpecConfiguration, name string, errs Errors) Errors {
+	if len(spec.Filters) == 0 {
+		errs.AddForProperty("proposal_submission.terms.change.new_market.changes.instrument.product.future."+name+".filters", ErrIsRequired)
+
+		return errs
+	}
+
+	for i, filter := range spec.Filters {
+		if filter.Key == nil {
+			errs.AddForProperty(fmt.Sprintf("proposal_submission.terms.change.new_market.changes.instrument.product.future."+name+".filters.%d.key", i), ErrIsNotValid)
```

Comment:
**Nit-pick / non-blocker**: You can add `continue` in the `if` and remove the `else` branch.